### ISSUE TITLE
Hexchat goes to "Not Responding" with ZNC and Windows 10 Pro x64

### ISSUE THREAD

**Issue by cshabazian** [hexchat/hexchat#2024]
Hexchat 2.12.4 on Windows 10 Pro (x64), not portable, x64 build.

When I attempt to connect to my znc server (version 1.6.5), hexchat connects to the server, then while trying to load the channels, it takes a VERY long time.  If I select a channel, hexchat goes into "Not Responding" state.  If I let it sit and connect to all the channels, it eventually connects, then once connected, immediately disconnects.  I can connect directly to the server my znc server connects to without a problem, and XChat connects to the znc server without problems, so the problem sure seems to be between hexchat and znc.

Looking for ways to troubleshoot this.

**Comment by hackitfast** [comment]
I'm having a similar issue. I am connected to a ZNC server with HexChat and it will intermittently go into a "Not Responding" state. I'm not sure exactly what triggers it, but I can supply some Windows logs.

I am also running Windows 10 Pro (x64), set to automatically connect to four servers on IRC at HexChat start.

[Report.txt](https://github.com/hexchat/hexchat/files/1206011/Report.txt)

**Comment by MayaPosch** [comment]
I got the same issue here. Windows 10 Pro x64. I can connect to any range of regular IRC networks just fine, but adding the one ZNC bouncer to the list causes HexChat to freeze up constantly, with only brief moments when it's responsive.

On Windows 7 I have the same server list and do not have these freezing issues, indicating that it's a problem only on Windows 10.

For now I'll have to find another IRC client to use as use of ZNC is mandatory.

**Comment by hackitfast** [comment]
@MayaPosch 

Try switching HexChat to Windows 7 compatibility mode for that issue. I am able to add ZNC servers on Windows 10 without any compatibility, the issue I am having is with random application hangs and crashes.

**Comment by cshabazian** [comment]
Definately a HexChat issue.  I work just fine with XChat, mIRC and adiirc under Win 10.

Changing to Windows 7 compatibility mode doesn't fix the problem.

From event viewer:
The program hexchat.exe version 2.12.4.0 stopped interacting with Windows and was closed. To see if more information about the problem is available, check the problem history in the Security and Maintenance control panel.

**Comment by MayaPosch** [comment]
Quick update: I'm having a similar issue with Hexchat (latest version, x64 version) on Windows 7. Trying to connect to the same ZNC bouncer which used to work fine until yesterday now results in the entire application freezing up, then eventually connecting, but not before all other networks Hexchat tries to connect to first time out.

**Comment by deviousgeek** [comment]
Having the same issue here (Windows 10 Pro x64). The same version on Linux (Debian Stretch x64) is working great.

**Comment by stevenroose** [comment]
Idem here. My Linux machine got stolen and needed to log into IRC after like 2 weeks and it seems that HexChat can't handle the dumps of channel buffers like WeeChat used to handle them.

**Comment by chowells79** [comment]
I've noticed this happens on both my desktop and my laptop, but my desktop usually recovers from the problem quickly, while the laptop loops a lot.  Since my desktop has way more cores and is quite a lot faster per core, this leads me to hypothesize that the issue is an accidentally-quadratic append somewhere in the ZNC code, along with some sort of timeout.

**Comment by HTlocal** [comment]
I seem to have the same issue with HexChat 2.12.4 on an Ubuntu 16.04 Linux 4.13.0-32-generic.
It connects to the server (I have automatic login enabled) and freezes the moment it opens the channels.
Safe mode doesn't help either.
Reading the other comments, I might add I'm on a laptop as well.

**Comment by HTlocal** [comment]
I fixed the issue by disabling all options in the logging tab of the preferences menu.

**Comment by hacknup** [comment]
Running linux and having same issue with HexChat 2.12.4.   Periodically disconnects when other clients do not.  It's like the net isn't there and drops the connect / reconnects.  Weird annoying issue.

Turning off all logging does not fix my issue.

Linux 4.15.6-300.fc27.x86_64 #1 SMP Mon Feb 26 18:43:03 UTC 2018 x86_64 GNU/Linux
HexChat 2.12.4

**Comment by cwebber** [comment]
(Hm, which doesn't seem to match what @hacknup said though, since they're running the version I rolled back to.)

**Comment by cwebber** [comment]
Update: I was wrong, it's still freezing with 2.12.4.  But I am connected to ZNC so I think it's still this same bug.

**Comment by dbedrenko** [comment]
This happens to me on Linux with Hexchat 2.14.1 . Absolutely unusuable. I tried starting with `--no-plugins`, and it didn't help. Nothing is shown in the command line output.

It freezes at startup for about 40 seconds, then repeatedly unlocks for a few minutes, then freezes for a minute, then unlocks etc.

I'm going back to XChat because this is ridiculous.

**Comment by wolfy1339** [comment]
It may have to do with the fact that znc is sending all buffer at once, and thus causes hexchat to crash because it can't keep up

**Comment by cshabazian** [comment]
And how would one test for that?  I'm happy to run any tests I can to help debug this issue

**Comment by wolfy1339** [comment]
I have no idea really, maybe try lowering the buffer until you find the sweet spot
________________________________

**Comment by cshabazian** [comment]
No joy.  lowered "Maximum Buffer Size" in the Global Settings to 100 and it has the same problem.

**Comment by hackitfast** [comment]
Yeah I'm on ZNC and am also having this bug. I've noticed that when you lock the computer (Win+L) HexChat will sometimes crash without telling you as well, unsure if other users connecting without ZNC are also having this issue.

**Comment by arch1mede** [comment]
WIN10 Hexchat 2.14.2 Portable and I am experiencing this bug. As its reading the buffers, I get Not Responding and it takes a good 1-2 mins then when its done reading the buffers, it will disconnect then reconnect.

**Comment by beren12** [comment]
still have this issue on latest release

**Comment by treeshateorcs** [comment]
+1 same here

as soon as i type a letter it hangs

**Comment by Starhowl** [comment]
I too am having this issue; running HexChat 2.14.3 on Windows 10 Professional with ZNC makes it almost every single time hangs at startup.
